### PR TITLE
ADTS-53 Changing Monaco Editor into Main Component

### PR DESCRIPTION
**Pull request by JoseZarmada**
All the changes are address
-Visuals for Monaco and Current Interface similarities.
-Multiline Functionality as implemented before

Also developing this notice the Intellisence code is Missing from the monaco-editor branch but the merge of the previos PR was executed succesfully.

### CODE REVIEW

File: client/src/components/QueryComponent/QueryComponent.js [hunk 1]
```diff
@@ -81,17 +66,34 @@ class QueryComponent extends Component {
   }
 
   onChange = evt => {
-    this.setState({ selectedQuery: evt.target.value, selectedQueryKey: null });
+    this.handleEditorChange(evt);
+    this.setState({ selectedQuery: evt, selectedQueryKey: null });
```
Comment (jamescarpinter) [inline client/src/components/QueryComponent/QueryComponent.js]: Is evt an event object? Or is it a string? If it's the latter, then we should probably change the name. If it is an event, then... why is it set to a string in the constructor?

File: client/src/components/QueryComponent/QueryComponent.js [hunk 2]
```diff
@@ -81,17 +66,34 @@ class QueryComponent extends Component {
...
+      this.executeQuery(event);
+    } else {
+      const evt = new Event(event.type, event);
+      event.target.dispatchEvent(evt);
+    }
+  }
+
+  onBlur = evt => {
+    this.setState({ selectedQueryHold: evt.target.value, selectedQuery: evt.target.value.replaceAll("\n", " ") });
+    this.handleEditorChange(evt.target.value.replaceAll("\n", " "));
```
Comment (jamescarpinter) [inline client/src/components/QueryComponent/QueryComponent.js]: Let's create a local variable with the rsult of the replace function and then use it in both places.

File: client/src/components/QueryComponent/QueryComponent.js [hunk 3]
```diff
@@ -31,17 +31,15 @@ class QueryComponent extends Component {
     this.state = {
       queries: [],
       selectedQuery: defaultQuery,
-      selectedQueryMultiline: "",
+      selectedQueryHold: defaultQuery,
```
Comment (jamescarpinter) [inline client/src/components/QueryComponent/QueryComponent.js]: Maybe `selectedQueryWithNewlines` or similar would be a better name for this state property?

File: client/src/components/QueryComponent/QueryComponent.js [hunk 4]
```diff
@@ -81,17 +66,34 @@ class QueryComponent extends Component {
...
+    const { selectedQueryHold } = this.state;
+    this.setState({ selectedQuery: selectedQueryHold });
+    this.handleEditorChange(selectedQueryHold);
   }
 
-  handleEditorBlur = evt => {
-    this.setState({ displayEditor: false, selectedQuery: evt.target.value.replaceAll("\n", " ") });
+  handleEditorChange = value => {
+    const count = Math.min(20, Math.max(0, value.split("").filter(c => c === "\n").length));
+    this.setState({ rowHeight: `${((count + 1) * 19) + 10}px` });
```
Comment (jamescarpinter) [inline client/src/components/QueryComponent/QueryComponent.js]: Can we make some of this values constants? I.e. so it's clear what they represent.

E.g. `const maxNumberOfRows = 20;` or `const lineHeight = 19;`.

File: client/src/components/QueryComponent/QueryComponent.scss [hunk 5]
```diff
@@ -303,15 +303,34 @@
...
+  border-color:rgb(208, 208, 208);
+  border-width:1px;
+  border-style: solid;
+  box-shadow: none;
+  box-sizing:border-box;
+  padding: 0px;
 }
 
+.monaco-editor-background {
+  background-color: var(--input-color)  !important;
```
Comment (jamescarpinter) [inline client/src/components/QueryComponent/QueryComponent.scss]: I would really prefer that we didn't use `!important`. Could we make the CSS selector more specific so we can override properly? Additional classes in the selector will make it more specific and help the style "win out".

**Review by jamescarpinter — CHANGES_REQUESTED** [review]

File: client/src/components/QueryComponent/QueryComponent.js (hunk 1 above)
Comment (JoseZarmada) [reply client/src/components/QueryComponent/QueryComponent.js]: It is supposed to be an event but the Monaco Editor exectute the function and send the value as a String so I Change the name to selectedQuery

File: client/src/components/QueryComponent/QueryComponent.js (hunk 2 above)
Comment (JoseZarmada) [reply client/src/components/QueryComponent/QueryComponent.js]: Done

File: client/src/components/QueryComponent/QueryComponent.js (hunk 3 above)
Comment (JoseZarmada) [reply client/src/components/QueryComponent/QueryComponent.js]: Done

File: client/src/components/QueryComponent/QueryComponent.js (hunk 4 above)
Comment (JoseZarmada) [reply client/src/components/QueryComponent/QueryComponent.js]: Done

File: client/src/components/QueryComponent/QueryComponent.scss (hunk 5 above)
Comment (JoseZarmada) [reply client/src/components/QueryComponent/QueryComponent.scss]: Done

File: client/src/components/QueryComponent/QueryComponent.js [hunk 6]
```diff
@@ -81,19 +81,22 @@ class QueryComponent extends Component {
...
+    this.handleEditorChange(selectedQueryWithNewlines);
   }
 
   handleEditorChange = value => {
-    const count = Math.min(20, Math.max(0, value.split("").filter(c => c === "\n").length));
-    this.setState({ rowHeight: `${((count + 1) * 19) + 10}px` });
+    const maxNumberOfRows = 20;
+    const lineHeight = 19;
+    const count = Math.min(maxNumberOfRows, Math.max(0, value.split("").filter(c => c === "\n").length));
+    this.setState({ rowHeight: `${((count + 1) * lineHeight) + 10}px` });
```
Comment (jamescarpinter) [inline client/src/components/QueryComponent/QueryComponent.js]: Can we define what 10 is as well? Looks like some kind of min height maybe?

**Review by jamescarpinter — CHANGES_REQUESTED** [review]

**Review by jamescarpinter — APPROVED** [review]

File: client/src/components/QueryComponent/QueryComponent.js (hunk 6 above)
Comment (JoseZarmada) [reply client/src/components/QueryComponent/QueryComponent.js]: Done

**Pull request merged**